pinot-controller/src/main/java/org/apache/pinot/controller/workload/scheme/PropagationUtils.java

```diff
@@ -201,11 +291,51 @@ public static Set<QueryWorkloadConfig> getQueryWorkloadConfigsForTags(
     return matchedConfigs;
   }
 
+  /**
+   * Returns all possible Helix tags for a given tenant name.
+   *
+   * <p>This includes broker, offline, and realtime tags.</p>
+   *
+   * @param tenantName Tenant name.
+   * @return A list of Helix tags for the tenant.
+   */
   private static List<String> getAllPossibleHelixTagsFor(String tenantName) {
     List<String> helixTags = new ArrayList<>();
     helixTags.add(TagNameUtils.getBrokerTagForTenant(tenantName));
     helixTags.add(TagNameUtils.getOfflineTagForTenant(tenantName));
     helixTags.add(TagNameUtils.getRealtimeTagForTenant(tenantName));
     return helixTags;
   }
+
+  /**
+   * Extracts all top-level cost IDs from the given propagation scheme.
```

Comment:
Can we just call this propagationEntities and say it will not fetch for overrides?? top level cost IDs seems be a new nomenclature we are introducing here.